### PR TITLE
Feat/support clean

### PR DESCRIPTION
**Pull request by FredericEspiau**
MANAGER-2710

### CODE REVIEW

File: packages/manager/modules/support/src/models/ticket/ticket.service.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash';
```
Comment (antleblanc) [inline packages/manager/modules/support/src/models/ticket/ticket.service.js]: Prefer import needed method:

```js
// example
import map from 'lodash/map';
```

File: packages/manager/modules/support/src/support.controller.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import _ from 'lodash';
```
Comment (antleblanc) [inline packages/manager/modules/support/src/support.controller.js]: Prefer import only needed methods (refer to my previous comment).

File: packages/manager/modules/support/src/ticket/ticket.routing.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import _ from 'lodash';
```
Comment (antleblanc) [inline packages/manager/modules/support/src/ticket/ticket.routing.js]: Prefer import only needed methods (refer to my previous comment).

File: packages/manager/modules/support/src/tickets/tickets.service.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import _ from 'lodash';
```
Comment (antleblanc) [inline packages/manager/modules/support/src/tickets/tickets.service.js]: Prefer import only needed methods (refer to my previous comment).

File: packages/manager/apps/support/index.routing.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  abstract: true,
+  name: 'app',
+  redirectTo: 'app.account',
+};
+
+export const APP_ACCOUNT_STATE = {
+  name: 'app.account',
+  redirectTo: 'app.account.support',
+  template,
+};
```
Comment (marie-j) [inline packages/manager/apps/support/index.routing.js]: Imo, we should try avoid multiple state declaration in a file

File: packages/manager/modules/support/package.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    "dev": "rollup -c --environment BUILD:development",
+    "dev:watch": "yarn run dev --watch",
+    "prepare": "yarn run build",
+    "start": "lerna exec --stream --scope='@ovh-ux/manager-support' --include-filtered-dependencies -- yarn run build",
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-support' --include-filtered-dependencies -- yarn run dev",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-support' --include-filtered-dependencies -- yarn run dev:watch"
+  },
+  "dependencies": {
+    "bootstrap": "^3.4.1",
+    "bootstrap4": "twbs/bootstrap#v4.0.0",
```
Comment (marie-j) [inline packages/manager/modules/support/package.json]: Maybe we should only use `bootstrap4` (and the latest version)

File: packages/manager/modules/support/src/index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+const moduleName = 'ovhManagerSupportLazyLoading';
+
+angular
+  .module(moduleName, [
+    'ovhManagerCore',
+    'oc.lazyLoad',
+    'ui.router',
+  ])
+  .config(/* @ngInject */ ($stateProvider) => {
+    $stateProvider.state(`${state.name}.**`, {
```
Comment (marie-j) [inline packages/manager/modules/support/src/index.js]: I find this declaration less explicit that using the state name directly. I need to go to the routing file to get the mapping.

File: packages/manager/modules/support/src/index.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        const $ocLazyLoad = $transition$.injector().get('$ocLazyLoad');
+
+        return import('./support.module')
+          .then(mod => $ocLazyLoad.inject(mod.default || mod));
+      },
+    });
+  })
+  .run(/* @ngTranslationsInject:json ./translations */)
+  .run(/* @ngInject */ ($transitions, $translate) => {
+    $transitions.onBefore({ to: `${state.name}.**` }, () => $translate.refresh());
```
Comment (marie-j) [inline packages/manager/modules/support/src/index.js]: This should probably be global to all the apps

File: packages/manager/modules/support/src/models/ticket/message/message.class.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { FROM } from './message.constants';
+
+export default class {
```
Comment (marie-j) [inline packages/manager/modules/support/src/models/ticket/message/message.class.js]: 👍

File: packages/manager/modules/support/src/support.controller.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+  $onInit() {
+    const transitionCriteria = {
+      from: `${STATE_NAME}.**`,
+    };
+
+    this.unsubscribers = [
+      this.$scope.$on(EVENTS.startLoading, () => this.startLoading()),
+      this.$scope.$on(EVENTS.stopLoading, () => this.stopLoading()),
+      this.$transitions.onStart(transitionCriteria, () => this.startLoading()),
```
Comment (marie-j) [inline packages/manager/modules/support/src/support.controller.js]: This should not be needed, you can add @ovh-ux/ng-line-progress to your app and an error page module will be available soon https://github.com/ovh-ux/manager/pull/1209

File: packages/manager/modules/support/src/ticket/ticket.scss [hunk 11]
```diff
@@ -0,0 +1,7 @@
+.message {
+  background-color: transparent;
+  border: none;
+  margin: 0;
+  padding: 0;
+  word-break: normal;
```
Comment (marie-j) [inline packages/manager/modules/support/src/ticket/ticket.scss]: I think all those can be handled by bootstrap utilities

File: packages/manager/modules/support/src/tickets/tickets.controller.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    $state,
+    OtrsPopupService,
+  ) {
+    this.$rootScope = $rootScope;
+    this.$state = $state;
+    this.OtrsPopupService = OtrsPopupService;
+  }
+
+  $onInit() {
+    this.$rootScope.$on('ticket.otrs.reload', this.reload.bind(this, { cleanCache: true }));
```
Comment (marie-j) [inline packages/manager/modules/support/src/tickets/tickets.controller.js]: Why is this needed ?

File: packages/manager/modules/support/src/models/ticket/message/message.class.js (hunk 9 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/models/ticket/message/message.class.js]: Noticed that sometimes this file is named as `TitleCase` (eg `packages/manager/modules/pci/src/projects/Project.class.js`).

Should we have to rename it as `Message.class.js` instead? (in order to Harmonize things)

File: packages/manager/modules/support/src/tickets/tickets.controller.js (hunk 12 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/tickets/tickets.controller.js]: It has been implemented as is. We plan to removed it for the second part of the refactor.

This part can be then omit (or we can add a comment at least if you prefer).

File: packages/manager/modules/support/package.json (hunk 6 above)
Comment (antleblanc) [reply packages/manager/modules/support/package.json]: `"bootstrap": "^3.4.1",` has been removed ;)

File: packages/manager/modules/support/src/models/ticket/message/message.class.js (hunk 9 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/models/ticket/message/message.class.js]: Could be addressed later since this pull request receive the "quality check ok" label.

File: packages/manager/modules/support/src/support.controller.js (hunk 10 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/support.controller.js]: Since this ticket already  received the "quality check ok" label and #1209 isn't yet available, it could be addressed later on.

Thanks for the inputs.

File: packages/manager/modules/support/src/ticket/ticket.scss (hunk 11 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/ticket/ticket.scss]: We plan to address this in the second wave. Thanks for the inputs!

File: packages/manager/apps/support/index.routing.js (hunk 5 above)
Comment (antleblanc) [reply packages/manager/apps/support/index.routing.js]: We plan to address this in the second wave. Thanks for the inputs!

File: packages/manager/modules/support/src/index.js (hunk 7 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/index.js]: We plan to address this in the second wave. Thanks for the inputs!

File: packages/manager/modules/support/src/index.js (hunk 8 above)
Comment (antleblanc) [reply packages/manager/modules/support/src/index.js]: We plan to address this in the second wave. Thanks for the inputs!

**Comment by FredericEspiau** [comment]
Fabulous, whoever wrote it should get a raise

File: packages/manager/apps/support/index.js [hunk 13]
```diff
@@ -0,0 +1,29 @@
+import support from '@ovh-ux/manager-support';
+import ngOvhOtrs from '@ovh-ux/ng-ovh-otrs';
+import uiRouter from '@uirouter/angularjs';
+import angular from 'angular';
+
+/* eslint-disable import/no-webpack-loader-syntax, import/extensions */
+import 'script-loader!jquery';
+import 'script-loader!lodash';
```
Comment (cbourgois) [inline packages/manager/apps/support/index.js]: Is it required as you import lodash functions one by one ?

Comment (FredericEspiau) [reply packages/manager/apps/support/index.js]: Let's find out 💪

File: packages/manager/modules/support/src/ticket/index.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+const moduleName = 'ovhManagerSupportTicketLazyLoading';
+
+angular
+  .module(moduleName, [
+    'ovhManagerCore',
+    'oc.lazyLoad',
+    'ui.router',
+  ])
+  .config(/* @ngInject */ ($stateProvider) => {
+    $stateProvider.state(`${state.name}.**`, {
```
Comment (cbourgois) [inline packages/manager/modules/support/src/ticket/index.js]: Why do you use constants for state name ?
IMO it make the code more complicated to read.

File: packages/manager/modules/support/src/ticket/index.js [hunk 15]
```diff
@@ -0,0 +1,29 @@
+import angular from 'angular';
```
Comment (cbourgois) [inline packages/manager/modules/support/src/ticket/index.js]: `ticket` is not a child of `tickets` ? States should follow the inheritance.

File: packages/manager/apps/support/index.js [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+  APP_ACCOUNT_STATE,
+} from './index.routing';
+
+angular
+  .module('supportApp', [
+    ngOvhOtrs,
+    support,
+    uiRouter,
+  ])
+  .config(/* @ngInject */ ($stateProvider) => {
```
Comment (cbourgois) [inline packages/manager/apps/support/index.js]:
```suggestion
  .config(routing);
```

and add at the top `import routing from './support.routing';`

File: packages/manager/apps/web/client/app/app.js [hunk 17]
```diff
@@ -449,9 +446,6 @@ angular
     set(editableThemes, 'default.submitTpl', ['<button style="background:none;border:none" type="submit">', '<i class="fa fa-check green"></i>', '</button>'].join(''));
     set(editableThemes, 'default.cancelTpl', ['<button style="background:none;border:none" ng-click="$form.$cancel()">', '<i class="fa fa-times red"></i>', '</button>'].join(''));
   })
-  .config((OtrsPopupProvider, constants) => {
```
Comment (cbourgois) [inline packages/manager/apps/web/client/app/app.js]: If `constants .REDIRECT_URLS.listTicket` is not used anymore you could remove it from constants.

File: packages/manager/modules/support/src/support.routing.js [hunk 18]
```diff
@@ -0,0 +1,24 @@
+import component from './support.component';
+
+import { state as stateToTickets } from './tickets/tickets.routing';
+
+import { STATE_NAME } from './support.constants';
+
+export const state = {
+  name: STATE_NAME,
+  redirectTo: stateToTickets.name,
+  translations: { format: 'json', value: ['.'] },
```
Comment (cbourgois) [inline packages/manager/modules/support/src/support.routing.js]: I think you could remove it since you add a lot of `.run(/* @ngTranslationsInject:json ./translations */)` and you refresh the translations `onBefore` transitions (https://github.com/ovh-ux/manager/pull/1215/files#diff-10edda547bcaf6534e1f95d95717342eR30)

File: packages/manager/apps/support/index.js (hunk 16 above)
Comment (cbourgois) [reply packages/manager/apps/support/index.js]: Why do we need to configure states to test the support module ?

File: packages/manager/modules/support/src/models/ticket/message/translations/Messages_fr_FR.json [hunk 19]
```diff
@@ -0,0 +1,4 @@
+{
+  "ovhManagerSupport_ticket_message_from_customer": "Vous",
```
Comment (cbourgois) [inline packages/manager/modules/support/src/models/ticket/message/translations/Messages_fr_FR.json]:
```suggestion
  "support_ticket_message_from_customer": "Vous",
```

File: packages/manager/modules/support/src/ticket/ticket.routing.js [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+    reload: /* @ngInject */ $state => params => $state
+      .transitionTo(
+        $state.current,
+        { ...$state.params, ...params },
+        { reload: true, inherit: false },
+      ),
+    ticket: /* @ngInject */ ($transition$, ticketMessageService, ticketService) => ticketService
+      .get($transition$.params().id)
+      .then(ticketFromApi => ticketService.buildFromApi(ticketFromApi))
+      .then(ticket => ticketMessageService
```
Comment (cbourgois) [inline packages/manager/modules/support/src/ticket/ticket.routing.js]: Should be a one call method.

File: packages/manager/modules/support/src/ticket/helpers/ticket.linky.js [hunk 21]
```diff
@@ -0,0 +1,85 @@
+import angular from 'angular';
```
Comment (cbourgois) [inline packages/manager/modules/support/src/ticket/helpers/ticket.linky.js]: IMO it is not a good idea to copy/paste/modify.

File: packages/manager/modules/support/src/models/ticket/index.js [hunk 22]
```diff
@@ -0,0 +1,20 @@
+import angular from 'angular';
```
Comment (cbourgois) [inline packages/manager/modules/support/src/models/ticket/index.js]: Should be in `ticket` folder.

File: packages/manager/modules/support/src/models/ticket/message/translations/Messages_fr_FR.json (hunk 19 above)
Comment (cbourgois) [reply packages/manager/modules/support/src/models/ticket/message/translations/Messages_fr_FR.json]: As discussed together, it could stay like that.

File: packages/manager/apps/support/index.js [hunk 23]
```diff
@@ -0,0 +1,20 @@
...
+import 'script-loader!jquery';
+/* eslint-enable import/no-webpack-loader-syntax, import/extensions */
+
+import { registerState } from './index.routing';
+
+angular
+  .module('supportApp', [
+    ovhUxManagerSupport,
+    ovhUxNgOvhOtrs,
+    uiRouterAngularJs,
```
Comment (marie-j) [inline packages/manager/apps/support/index.js]: Sounds a bit obvious as we're in an `angular.module` declaration, don't you think ?

File: packages/manager/apps/support/index.js [hunk 24]
```diff
@@ -0,0 +1,20 @@
+import ovhUxManagerSupport from '@ovh-ux/manager-support';
+import ovhUxNgOvhOtrs from '@ovh-ux/ng-ovh-otrs';
```
Comment (marie-j) [inline packages/manager/apps/support/index.js]:
```suggestion
import ngOvhOtrs from '@ovh-ux/ng-ovh-otrs';
```
could do I think,

File: packages/manager/modules/support/src/support.controller.js [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+    );
+
+    this.$scope.$on(
+      EVENT_NAMES.stopLoading,
+      () => this.stopLoading(),
+    );
+  }
+
+  startLoading() {
+    this.isLoading = true;
```
Comment (marie-j) [inline packages/manager/modules/support/src/support.controller.js]: Imo, we should uniformize and use `ng-ui-router-line-progress` instead

File: packages/manager/modules/support/src/support.routing.js [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+export const state = {
+  name: 'support',
+  redirectTo: 'support.tickets',
+  url: '/support',
+  views: {
+    '@': component.name,
+  },
+};
+
+export const registerState = /* @ngInject */ ($stateProvider) => {
```
Comment (marie-j) [inline packages/manager/modules/support/src/support.routing.js]: Looks like a duplicate from another route declaration. I don't really see the point of using a function if it's duplicated all over routing files

File: packages/manager/modules/support/package.json [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+    "build": "rollup -c --environment BUILD:production",
+    "dev": "rollup -c --environment BUILD:development",
+    "dev:watch": "yarn run dev --watch",
+    "prepare": "yarn run build",
+    "start": "lerna exec --stream --scope='@ovh-ux/manager-support' --include-filtered-dependencies -- yarn run build",
+    "start:dev": "lerna exec --stream --scope='@ovh-ux/manager-support' --include-filtered-dependencies -- yarn run dev",
+    "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-support' --include-filtered-dependencies -- yarn run dev:watch"
+  },
+  "dependencies": {
+    "bootstrap4": "twbs/bootstrap#v4.0.0",
```
Comment (marie-j) [inline packages/manager/modules/support/package.json]: Could be renamed to `bootstrap` then ?

File: packages/manager/modules/support/src/tickets/ticket/message/message.scss [hunk 28]
```diff
@@ -0,0 +1,7 @@
+.message {
+  background-color: transparent;
+  border: none;
+  margin: 0;
+  padding: 0;
+  word-break: normal;
```
Comment (marie-j) [inline packages/manager/modules/support/src/tickets/ticket/message/message.scss]: This could be achieved with bootstrap utilities

File: packages/manager/modules/support/src/tickets/ticket/ticket.module.js [hunk 29]
```diff
@@ -0,0 +1,34 @@
+import angular from 'angular';
+
+import ovhUxManagerCore from '@ovh-ux/manager-core';
+import ovhUxNgTranslateAsyncLoader from '@ovh-ux/ng-translate-async-loader';
+import uiRouterAngularJs from '@uirouter/angularjs';
+import angularTranslate from 'angular-translate';
+import 'ovh-ui-angular';
+
+import 'ovh-ui-kit/dist/oui.css';
+import 'ovh-ui-kit-bs/dist/ovh-ui-kit-bs.css';
```
Comment (marie-j) [inline packages/manager/modules/support/src/tickets/ticket/ticket.module.js]: Why is it needed, it's all ui-kit components and styles that has been used, right ?

File: packages/manager/modules/support/src/tickets/tickets.module.js [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+    ovhUxNgTranslateAsyncLoader,
+    ticket,
+    uiRouterAngularJs,
+  ])
+  .component(component.name, component)
+  .config(registerState)
+  .run(/* @ngTranslationsInject:json ./translations */)
+  .service('ticketService', ticketService)
+  .service('ticketMessageService', ticketMessageService)
+  .service('ticketsService', ticketsService);
```
Comment (marie-j) [inline packages/manager/modules/support/src/tickets/tickets.module.js]: Looks like quite a few services here. Would it make sense to merge `ticketsService` with `ticketService` ? They both handle `Ticket` elements right

File: packages/manager/modules/support/src/support.controller.js (hunk 25 above)
Comment (FredericEspiau) [reply packages/manager/modules/support/src/support.controller.js]: This is used for form submission, not for state transition, this is what was advised by CX :)

File: packages/manager/apps/support/index.js (hunk 23 above)
Comment (FredericEspiau) [reply packages/manager/apps/support/index.js]: It's the name of the import, didn't wan't to have to think about the naming of my variable

File: packages/manager/apps/support/index.js (hunk 24 above)
Comment (FredericEspiau) [reply packages/manager/apps/support/index.js]: It wouldn't match the pattern for the other packages :(

File: packages/manager/modules/support/src/tickets/tickets.module.js (hunk 30 above)
Comment (FredericEspiau) [reply packages/manager/modules/support/src/tickets/tickets.module.js]: `tickets` works on the `tickets` view, `ticket` on the `Ticket` class
